datasets/chapterbreak.py

```diff
@@ -38,42 +39,72 @@ def split_in_pages(text: str, max_tokens_per_split: int) -> list[str]:
     return page_list
 
 
-def ordinal(n: int) -> str:
-    if 11 <= (n % 100) <= 13:
-        suffix = 'th'
-    else:
-        suffix = ['th', 'st', 'nd', 'rd', 'th'][min(n % 10, 4)]
-    return str(n) + suffix
+def deliver_in_pages(text: str, max_page_tokens: int, prefix: str = "") -> list[str]:
+    pages = split_in_pages(text, max_page_tokens)
+    if len(pages) == 1:
+        return [f"{prefix}:\n\n{pages[0]}"]
+    script = list()
+    last_i = len(pages) - 1
+    for i, page in enumerate(pages):
+        last_page_str = " and last" if i == last_i else ""
+        page_label = f"{ordinal(i + 1)}{last_page_str} page"
+        if prefix != "":
+            page_label = f"{prefix} ({page_label})"
+        script.append(f"{page_label}:\n\n{page}")
+    return script
 
 
 @dataclass
 class ChapterBreakDataset(DatasetInterface):
     name: str = "ChapterBreak"
     description: str = (
-        "The agent is given a text that corresponds to the final part of a book "
-        "chapter, plus 6 possible continuations. It is then asked to say which of the "
-        "six continuations is the right one."
+        "The agent is given a text that corresponds to the content of many book chapters, in addition to 6 possible "
+        "beginnings for the next chapter. It is then asked to say which of the six continuations is the right one."
     )
-    split: str = "ao3"  # pg19 / ao3 / all
-    page_tokens: int = 1024
-    reset_message: str = "Forget the current chapter and its potential continuations."
+    reset_message: str = "Forget the chapters that I have read you and the potential continuations that I gave to you."
+    # The GoodAI split is a selection of samples that have been inspected by us to ensure that they are solvable, also
+    # removing ordering hints from the chapter titles.
+    split: str = "goodai"  # goodai / pg19 / ao3 / all
+    selection_info: dict = field(default_factory=dict)
 
     def __post_init__(self):
-        assert self.split in {"pg19", "ao3", "all"}
+        assert self.split in {"goodai", "pg19", "ao3", "all"}
 
     def load_data(self) -> dict:
         filename = f"chapterbreak_ctx_8192.json"
         path = get_gdrive_file(self.name, GDRIVE_8K_ID, filename)
         with open(path) as fd:
             return json.load(fd)
 
+    def apply_sample_selection(self, samples: dict) -> dict:
+        with open(get_data_path(self.name, "chapterbreak-goodai-selection.json")) as fd:
+            selection = json.load(fd)
+        sample_selection = dict()
+        for sel in selection:
+            if not sel["solvable"]:
```

Comment:
We are not selecting unsolvable samples. So Why are they in the split file?

Reply:
I decided to keep them to have a record of what samples have been subjected to human scrutiny, in order to save efforts in the future.